src/Serilog.Sinks.AzureDataExplorer/Sinks/AzureDataExplorerDurableSink.cs

```diff
@@ -0,0 +1,187 @@
+                    ColumnName = m.ColumnName,
+                    ColumnType = m.ColumnType,
+                    Properties = new Dictionary<string, string>(1)
+                    {
+                        {
+                            MappingConsts.Path, m.ValuePath
+                        }
+                    }
+                }).ToList();
+            }
+            else
+            {
+                ingestionMapping.IngestionMappings = SDefaultIngestionColumnMapping;
+            }
+
+            var kcsb = options.GetKustoConnectionStringBuilder();
+            m_ingestClient = KustoIngestFactory.CreateQueuedIngestClient(kcsb);
+            m_sink = new LoggerConfiguration()
+                .MinimumLevel.Verbose()
+                .WriteTo.File(new CompactJsonFormatter(),
+                    options.BufferBaseFileName + FileNameSuffix,
+                    restrictedToMinimumLevel: LevelAlias.Minimum,
+                    fileSizeLimitBytes: options.BufferFileSizeLimitBytes,
+                    levelSwitch: options.BufferFileLoggingLevelSwitch,
+                    buffered: false,
+                    shared: false,
+                    flushToDiskInterval: TimeSpan.FromSeconds(10),
+                    rollingInterval: options.BufferFileRollingInterval,
+                    rollOnFileSizeLimit: true,
+                    retainedFileCountLimit: options.BufferFileCountLimit,
+                    encoding: Encoding.UTF8
+                ).CreateLogger();
+
+            var payloadReader = new AzureDataExplorerPayloadReader(
+                rollingInterval: options.BufferFileRollingInterval);
+
+            m_shipper = new LogShipper<List<LogEvent>>(
+                bufferBaseFilename: options.BufferBaseFileName,
+                batchPostingLimit: options.BatchPostingLimit,
+                period: options.BufferLogShippingInterval ?? TimeSpan.FromSeconds(5),
+                eventBodyLimitBytes: options.SingleEventSizePostingLimit,
+                levelControlSwitch: options.BufferFileLoggingLevelSwitch,
+                payloadReader: payloadReader,
+                bufferSizeLimitBytes: options.BufferFileSizeLimitBytes,
+                rollingInterval: options.BufferFileRollingInterval,
+                ingestClient: m_ingestClient,
+                formatProvider: formatProvider,
+                databaseName: databaseName,
+                tableName: tableName,
+                ingestionMapping: ingestionMapping,
+                flushImmediately: flushImmediately
+            );
+        }
+
+        public void Emit(LogEvent logEvent)
+        {
+            // This is a lagging indicator, but the network bandwidth usage benefits
+            // are worth the ambiguity.
+            if (m_shipper.IsIncluded(logEvent))
+            {
+                m_sink.Write(logEvent);
+            }
+        }
+
+        public void Dispose()
+        {
+            m_sink.Dispose();
+            m_shipper.Dispose();
```

Comment:
order of dispose , does this matter ?

Reply:
since both doesnt have a parent-child relationship, I dont think it should matter. Added ingestClient as well. Since ingestClient was implements disposable.